fdbserver/Ratekeeper.actor.cpp

```diff
@@ -1024,6 +1024,10 @@ void updateRate(RatekeeperData* self, RatekeeperLimits* limits) {
 
 	std::map<UID, limitReason_t> ssReasons;
 
+	bool printRateKeepLimitReasonDetails =
+	    (deterministicRandom()->random01() < SERVER_KNOBS->RATEKEEPER_LIMIT_REASON_SAMPLE_RATE) &&
+	    SERVER_KNOBS->RATEKEEPER_PRINT_LIMIT_REASON;
```

Comment:
put RATEKEEPER_PRINT_LIMIT_REASON before the random01(). 
when it's set to false, it can save the call to random01().